apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -189,88 +188,121 @@ def db_plugin(self):
+    def _bind_opflex_port(self, context, segment, agent):
+        """Populate VIF type and details for OpFlex VIFs.
+
+           For OpFlex VIFs, we just report the OVS VIF type,
+           along with security groups setting, which were
+           set when this mechanism driver was instantiated.
+        """
         if self._check_segment_for_agent(segment, agent):
-            vif_type = self.vif_type
-            vif_details = self.vif_details
-            if self._is_dvs_vif_type(context, agent):
-                vif_type = acst.VIF_TYPE_DVS
-                vif_details = self._get_dvs_vif_details(context)
-            context.set_binding(segment[api.ID], vif_type, vif_details)
+            context.set_binding(segment[api.ID],
+                                portbindings.VIF_TYPE_OVS,
+                                {portbindings.CAP_PORT_FILTER:
+                                 self.sg_enabled,
+                                 portbindings.OVS_HYBRID_PLUG:
+                                 self.sg_enabled})
             return True
         else:
             return False
 
     def _check_segment_for_agent(self, segment, agent):
-        network_type = segment[api.NETWORK_TYPE]
-        if network_type == ofcst.TYPE_OPFLEX:
-            opflex_mappings = agent['configurations'].get('opflex_networks',
-                                                          [])
+        """Check support for OpFlex type segments.
+
+           The agent has the ability to limit the segments in OpFlex
+           networks by specifying the mappings in their config. If no
+           mapping is specifified, then all OpFlex segments are
+           supported.
+        """
+        if segment[api.NETWORK_TYPE] == ofcst.TYPE_OPFLEX:
+            opflex_mappings = agent['configurations'].get('opflex_networks')
             LOG.debug(_("Checking segment: %(segment)s "
                         "for physical network: %(mappings)s "),
                       {'segment': segment, 'mappings': opflex_mappings})
             return (opflex_mappings is None or
                     segment[api.PHYSICAL_NETWORK] in opflex_mappings)
         else:
-            mappings = agent['configurations'].get('bridge_mappings', {})
-            tunnel_types = agent['configurations'].get('tunnel_types', [])
-            LOG.debug(_("Checking segment: %(segment)s "
-                        "for mappings: %(mappings)s "
-                        "with tunnel_types: %(tunnel_types)s"),
-                      {'segment': segment, 'mappings': mappings,
-                       'tunnel_types': tunnel_types})
-            if network_type == 'local':
-                return True
-            elif network_type in tunnel_types:
-                return True
-            elif network_type in ['flat', 'vlan']:
-                return segment[api.PHYSICAL_NETWORK] in mappings
-            else:
-                return False
+            return False
```

Comment:
I understand from Ivar that the opflex L2 agent no longer supports the 'flat', 'vlan' or various tunnel network types, so returning False for these makes sense. But I wouldn't be surprised if it did support the 'local' type. If so, we should probably return True for that. It probably also wouldn't be too difficult to make the DVS agent support 'local'. Since 'local' is rarely used, and maybe not currently supported, I think this can be addressed later.